spec/amp-cors-requests.md

```diff
@@ -49,3 +49,11 @@ The resulting HTTP response has to also contain the following headers:
  - `Access-Control-Allow-Origin: <origin>`. Here "origin" refers to the requesting origin that was allowed via "Origin" request header above. Ex: "https://cdn.ampproject.org". This is a CORS spec requirement. Notice that while CORS spec allows the value of '\*' to be returned in this header, AMP strongly discourages use of '\*'. Instead the value of the "Origin" header should be validated and echoed for improved security.
  - `AMP-Access-Control-Allow-Source-Origin: <source-origin>`. Here "source-origin" indicates the source origin that is allowed to read the authorization response as was verified via "__amp_source_origin" URL parameter. Ex: "https://publisher1.com".
  - `Access-Control-Expose-Headers: AMP-Access-Control-Allow-Source-Origin`. This header simply allows CORS response to contain the "AMP-Access-Control-Allow-Source-Origin" header.
+
+#### Note on non-idempotent Requests
+When making CORS requests that would change the state of your system (e.g. User subscribes to or unsubscribes from a mailing list) the first two steps you need to make do are:
+
+1. Check the `Origin` header. If the origin was not `*.ampproject.org` or the publisher's origin, stop and return an error response.
```

Comment:
Can you explain more why you want to do each of these things? For instance, with this one, is the point that the publisher should only process stuff from trusted places? How do we future-proof on additional caches being trusted?